sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnSchemaInformation.java

```diff
@@ -101,7 +107,16 @@ DoFnSchemaInformation withSelectFromSchemaParameter(
                     unbox))
             .build();
 
-    return toBuilder().setElementConverters(converters).build();
+    List<FieldAccessDescriptor> descriptors =
+        ImmutableList.<FieldAccessDescriptor>builder()
+            .addAll(getFieldAccessDescriptors())
+            .add(selectDescriptor)
+            .build();
+
+    return toBuilder()
+        .setElementConverters(converters)
+        .setFieldAccessDescriptors(descriptors)
+        .build();
```

Comment:
This is not always called (withUnboxFromPrimitiveParameter as called as well). I think better to make this a separate builder called from ParDo.java

Reply:
I refactored this so setElementConverters and setFieldAccessDescriptors are separate methods.